Sources/StreamChat/StateLayer/ChatState.swift

```diff
@@ -75,6 +77,28 @@ public final class ChatState: ObservableObject {
     public var isLoadingPreviousMessages: Bool {
         paginationState.isLoadingPreviousMessages
     }
+    
+    // MARK: - Message Reading
+    
+    /// The id of the message which the current user last read.
+    public var lastReadMessage: MessageId? {
+        guard let channel else { return nil }
+        guard let userId = authenticationRepository.currentUserId else { return nil }
+        return channel.lastReadMessageId(userId: userId)
+    }
+    
+    /// The id of the first unread message.
+    ///
+    /// The returned message id follows requirements:
+    /// * Read state is unavailable: oldest message if all the messages have been paginated, otherwise nil
+    /// * Unread message count is zero: nil
+    /// * Read state's ``ChatChannelRead.lastReadMessageId`` is nil: oldest message if all the messages have been paginated, otherwise nil
+    /// * Last read message is unreachable (e.g. channel was truncated): oldest message if all the messages have been paginated, otherwise nil
+    /// * Next message after the last read message id not from the current user
+    public var firstUnreadMessage: MessageId? {
```

Comment:
I did my best to document the logic we have for driving this since it gets complicated really fast.